druid/src/core.rs

```diff
@@ -189,34 +193,33 @@ impl<T, W: Widget<T>> WidgetPod<T, W> {
 
     /// Set the layout [`Rect`].
     ///
+    /// This is soft-deprecated; you should use [`set_origin`] instead for new code.
+    pub fn set_layout_rect(&mut self, ctx: &mut LayoutCtx, data: &T, env: &Env, layout_rect: Rect) {
+        if layout_rect.size() != self.state.size {
+            log::warn!("set_layout_rect passed different size than returned by layout method");
+        }
+        self.set_origin(ctx, data, env, layout_rect.origin());
+    }
+
+    /// Set the origin of this widget, in the parent's coordinate space.
+    ///
     /// A container widget should call the [`Widget::layout`] method on its children in
-    /// its own [`Widget::layout`] implementation, then possibly modify the returned [`Size`], and
-    /// finally call this `set_layout_rect` method on the child to set the final layout [`Rect`].
+    /// its own [`Widget::layout`] implementation, and then call `set_origin` to
+    /// position those children.
     ///
     /// The child will receive the [`LifeCycle::Size`] event informing them of the final [`Size`].
```

Comment:
Do we still need `LifeCycle::Size` if your size is always determined by your own layout function?

Reply:
I was wondering about this too. I guess it depends on how this event is actually used? I could see it making sense in some cases, because it means the widget doesn't need to internally hold on to past sizes. I'm not actually sure if this matters, though? I don't believe I use this event anywhere.

Reply:
I guess I can't imagine anything that I would want to do in `lifecycle` that I couldn't just do in `layout` :shrug: Given our lack of API stability, my inclination would be to get rid of it and then put it back later if someone finds a use. But I'll give you the green button anyway...

Reply:
Maybe @xStrom would like to chime in? Removing this would also let us remove some special-case code we use to send this notification..